### DISCUSSION TITLE
how to disable maximize button on a window ?

### DISCUSSION THREAD

**Discussion by ffred** [slint-ui/slint#5106] [Q&A]
Hi,
I think I already seen something about that, but can't find any info right now.. 
how can I remove the maximize button or disable it ?
how can I define a window to be non resizable ??
thanks

[Edit:] I set min-height/width and max-height/width and it's nice like that. it keeps the resize possibilities in something that suits me well.
but still got the "problem" that the maximize button goes over the max sizes I set, to full screen. and I don't want that. so, still need a way to disable it.....

**Comment by ffred** [comment]
I'm on Windows, so looked for info on Winit side and found something interesting [here](https://github.com/rust-windowing/winit/issues/947), but it doesn't seem to work.
I set min-height/width and max-height/width to same values but the maximize button is still there...

also found a [set_resizable](https://docs.rs/winit/latest/winit/window/struct.Window.html) function in Winit, so will see if I could use that. 
sorry, still beginning with Slint, so still searching how to do things..

**Reply by ogoffart** [reply]
You can only access winit with the private API: https://docs.rs/i-slint-backend-winit/latest/i_slint_backend_winit/
But using that, you should be able to do what it.

**Comment by ffred** [comment]
Thanks !

**Comment by Dovelong0713** [comment]
I want to know if you have implemented the disable maximize button, because sometimes I also need to disable the maximize minimize and close buttons  @ffred

**Comment by ffred** [comment]
Hi,
Sorry, I had to pause on this project to finish another one, with no UI.. so no, I didn't finished it for now

**Comment by ffred** [comment]
@Dovelong0713 
Hi, just did a rapid test to do it.
here is my  Cargo.toml file :
```
[package]
name = "tst-resize"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
slint = "1.5.1"
i-slint-backend-winit = "=1.5.1"
winit = "0.30.0"

[build-dependencies]
slint-build = "1.5.0"
```
here is my build.rs file :
```
fn main() {    
    slint_build::compile("ui/main.slint").unwrap();
}
```
my main.slint file :
```
export component HelloWorld inherits Window {
    width: 400px;
    height: 180px;

    Text {
       x: parent.width / 2 - 40px;
       y: parent.height / 2;
       text: "Hello, world";
       color: blue;
    }
}
```
and finally, my main.rs file :
```
#![windows_subsystem = "windows"]

use i_slint_backend_winit;
use i_slint_backend_winit::winit::window::WindowButtons;
use i_slint_backend_winit::WinitWindowAccessor;


slint::include_modules!();

fn main() {
    let helloworld = HelloWorld::new().unwrap();

    helloworld.window().with_winit_window(|winit_win: &i_slint_backend_winit::winit::window::Window| {
       winit_win.set_enabled_buttons(WindowButtons::MINIMIZE | WindowButtons::CLOSE);
    });

    helloworld.run().unwrap();
}
```
and I got a window with a desactivated maximize button !  :-)

**Reply by Dovelong0713** [reply]
Thank you very much, I need this implementation
